lib/ui/profitloss/RecordAmount.dart

```diff
@@ -0,0 +1,373 @@
+
+    listBuilder.add(
+      RecordAmountListItem(
+        viewModel: RecordAmountListItemViewModel(
+          type: RecordCellType.pickDate,
+          isEditable: viewModel.isEditable,
+          selectedDate: viewModel.isEditable
+              ? _initDate
+              : viewModel.actions[_Constants.firstListItem].selectedDate,
+        ),
+        parent: this,
+      ),
+    );
+
+    listBuilder.add(ListDivider.build());
+
+    listBuilder.add(
+      RecordAmountListItem(
+        viewModel: RecordAmountListItemViewModel(
+          type: RecordCellType.pickItem,
+          isEditable: viewModel.isEditable,
+          selectedItem: viewModel.isEditable
+              ? selectedCrop
+              : viewModel.actions[_Constants.secondListItem].selectedItem,
+          listOfCrops: viewModel.actions[_Constants.secondListItem].listOfCrops,
+        ),
+        parent: this,
+      ),
+    );
+
+    listBuilder.add(ListDivider.build());
+
+    listBuilder.add(
+      RecordAmountListItem(
+        viewModel: RecordAmountListItemViewModel(
+          type: RecordCellType.description,
+          isEditable: viewModel.isEditable,
+          description: viewModel.isEditable
+              ? description
+              : viewModel.actions[_Constants.thirdListItem].description,
+        ),
+        style: RecordAmountListItemStyles.biggerStyle,
+        parent: this,
+      ),
+    );
+
+    if (viewModel.isEditable) {
+      listBuilder.add(_buildFooter(viewModel, style));
+    }
+    return listBuilder;
+  }
+
+  Padding _buildFooter(
+      RecordAmountViewModel viewModel, RecordAmountStyle style) {
+    return Padding(
+      padding: style.buttonEdgePadding,
+      child: Row(
+        children: <Widget>[
+          Expanded(
+            child: !widget._viewModel.isFilled
+                ? RoundedButton(
+                    viewModel: RoundedButtonViewModel(
+                        title: viewModel.buttonTitle, onTap: null),
+                    style:
+                        RoundedButtonStyle.largeRoundedButtonStyle().copyWith(
+                      backgroundColor: Color(0xFFe9eaf2),
+                    ),
+                  )
+                : RoundedButton(
+                    viewModel: RoundedButtonViewModel(
+                        title: viewModel.buttonTitle, onTap: saveData),
+                    style: viewModel.type == RecordType.sale
+                        ? RoundedButtonStyle.largeRoundedButtonStyle().copyWith(
+                            backgroundColor: Color(0xFF24d900),
+                          )
+                        : RoundedButtonStyle.largeRoundedButtonStyle().copyWith(
+                            backgroundColor: Color(0xFFff8d4f),
+                          ),
+                  ),
+          ),
+        ],
+      ),
+    );
+  }
+
+  checkIfFilled() {
+    setState(() {
+      if (amountIsFilled && cropIsFilled) {
+        widget._viewModel.isFilled = true;
```

Comment:
you can simplify the if:

`widget._viewModel.isFilled = amountIsFilled & cropIsFilled`